### PR TITLE
create_user_model

### PR DESCRIPTION
**Pull request by sergiiss**

### CODE REVIEW

File: app/assets/javascripts/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (kavalenka) [inline app/assets/javascripts/users.coffee]: Do we really need this empty file?

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (kavalenka) [inline app/assets/stylesheets/users.scss]: One more empty file.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to @user, notice: 'Пользователь был успешно создан.'
```
Comment (kavalenka) [inline app/controllers/users_controller.rb]: What about developing service in english language? Later we'll make multi language version :)

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (kavalenka) [inline app/helpers/users_helper.rb]: Do we need this empty file?

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+  validates :name, :length => {:maximum => 12}
+  validates :password, :length => {:in => 6..15}
+
```
Comment (kavalenka) [inline app/models/user.rb]: I think email should be validated too.

File: app/views/home_page/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  <div class="central-inner">
+    <div class="central-text">
+      <div class="cental-text-top">Built for developers</div>
+      <div class="central-text-bottom">
+        <p>
+          GitHub is a development platform inspired by the way you work. From open source to business, you can host and review code,   manage projects, and build software alongside millions of other developers.
+        </p>
+      </div>
+    </div>
+    <div class="input-form">
```
Comment (kavalenka) [inline app/views/home_page/index.html.erb]: That's be better to move form logic into a partial.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -8,9 +8,12 @@
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">
   </head>
 
   <body>
+    <%= render 'top' %>
```
Comment (kavalenka) [inline app/views/layouts/application.html.erb]: Please use here ```render partial```.

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -0,0 +1,96 @@
+# This file was generated by the `rails generate rspec:install` command. Conventionally, all
```
Comment (kavalenka) [inline spec/spec_helper.rb]: Please remove all this commented lines.

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb (hunk 7 above)
Comment (sergiiss) [reply app/views/layouts/application.html.erb]: Also note that explicitly specifying :partial is required when passing additional options such as :layout.
This is from the documentation on Rails.

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  validates :name, :email, presence: true, uniqueness: true
+  validates :email, length: { minimum: 4}
```
Comment (kavalenka) [inline app/models/user.rb]: Please add uniqueness option

File: spec/rails_helper.rb [hunk 10]
```diff
@@ -0,0 +1,57 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
```
Comment (kavalenka) [inline spec/rails_helper.rb]: One more file with comments. Please delete them.

File: spec/spec_helper.rb [hunk 11]
```diff
@@ -0,0 +1,12 @@
+RSpec.configure do |config|
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+
```
Comment (kavalenka) [inline spec/spec_helper.rb]: Remove empty line please.

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 9 above)
Comment (sergiiss) [reply app/models/user.rb]: I didn`t understand the comment. Uniqueness is already realized for name and email.

Comment (kavalenka) [reply app/models/user.rb]: My misunderstanding.
It's better to write validation for each attribute starting from a new line.
Like
``validates :name, presence: true, uniqueness: true``
``validates :email, presence: true, uniqueness: true, length: { minimum: 4}``

Comment (sergiiss) [reply app/models/user.rb]: Ok, I do it in the evening.

**Comment by kavalenka** [comment]
Great! Now let's squash all commits into new one and give some sensible name to pull request.

**Pull request closed without merging**